core/src/luigi-config/init-params.js

```diff
@@ -18,60 +27,55 @@ function getResponseParams(usePKCE = true) {
   }
 }
 
-function createSystemNamespacesList(namespaces) {
-  return namespaces ? namespaces.split(' ') : [];
-}
-
 export async function saveInitParamsIfPresent() {
-  const DEFAULT_MODULES = {
-    SERVICE_CATALOG: 'servicecatalog.k8s.io',
-    SERVICE_CATALOG_ADDONS: 'servicecatalog.kyma-project.io',
-    EVENTING: 'eventing.kyma-project.io',
-    API_GATEWAY: 'gateway.kyma-project.io',
-    APPLICATIONS: 'applicationconnector.kyma-project.io',
-    ADDONS: 'addons.kyma-project.io',
-    SERVERLESS: 'serverless.kyma-project.io',
-    SERVERLESS_REPOS: 'gitrepositories.serverless.kyma-project.io',
-  };
+  const encodedParams = new URL(location).searchParams.get('init');
+  if (encodedParams) {
+    const decoded = await encoder.decompress(encodedParams);
+
+    const isKubeconfigPresent = !!Object.keys(decoded.kubeconfig || {}).length;
+    const kubeconfigUser =
+      decoded.kubeconfig?.users && decoded.kubeconfig?.users[0].user;
+    const isOidcAuthPresent = decoded.config?.auth;
+
+    const requireMoreInput =
+      !isKubeconfigPresent ||
+      (!isOidcAuthPresent && !hasKubeconfigAuth(kubeconfigUser));
+
+    if (requireMoreInput) {
+      navigateToAddCluster(encodedParams);
+      return;
+    }
 
-  const initParams = new URL(location).searchParams.get('init');
-  if (initParams) {
-    const decoded = await encoder.decompress(initParams);
-    const systemNamespaces = createSystemNamespacesList(
-      decoded.config?.systemNamespaces,
-    );
     const params = {
       ...decoded,
       config: {
         ...decoded.config,
-        systemNamespaces,
         modules: { ...DEFAULT_MODULES, ...(decoded.config?.modules || {}) },
       },
+      currentContext: {
+        cluster: decoded?.kubeconfig?.clusters[0],
```

Comment:
If someone would accidentally add an incorrect kubeconfig object where for example the `clusters`/`users` field doesn't exist, then there is a blank page, no message, no redirection, no info for the user.
Probably, the `Luigi.ux().showAlert()` is called too early for Luigi.